### DISCUSSION TITLE
Get info about window that will be focused after closing current one

### DISCUSSION THREAD

**Discussion by jbriales** [i3/i3#4803] [Q&A] [answered]
When we close a window/container in a workspace, the focus seems to go to the last window/container that had the focus in that workspace, even if we focus other containers in different workspaces meanwhile.

My question is: Is there any way to query i3 about which window will be focused when we kill/close the current one, BEFORE we do that kill? Or more generally, is there a way to query the potential queue of windows to focus if we close one after another?

**Comment by Airblader** [comment, marked as answer]
> Or more generally, is there a way to query the potential queue of windows to focus if we close one after another?

In the tree response you'll find a `focus` property with a list of the children containers. That list is ordered by recency in terms of focus within the parent. That means, starting from your container, look at the parent's focus list and choose the one prior to this one. If it's the only child, recurse upwards and check the parent container itself.
